### ISSUE TITLE
test suite

### ISSUE THREAD

**Issue by glaszig** [mattheworiordan/capybara-screenshot#173]
just spent an hour debugging the test suite. i managed to resolve bundler dependency errors for each gemfile but what really holds things back is the requirement for old aruba versions in combination with older rspec and cucumber versions. these carry dependencies which can only be satisfied by aruba < 0.8.

but your changes in 3f80807 and in aruby 0.14 make it really hard to stay backward-compatible to cucumber 1.2, 1.3 and rspec 2.14. unless you have a better idea i'd suggest just dropping these.

**Comment by mattheworiordan** [comment]
Thanks @glaszig for looking at this, I appreciate the help as I am quite busy right now :(

So my concern is that Cucumber 2.0 was only released in March 2015, so not that long ago.  Not knowing that we're supporting 1.3 (released 2013 but kept going until Jun 2015) does worry me a bit.

RSpec on the other hand I am less concerned about. 2.14 was last published in July 2013, so that's old enough.

@glaszig what specifically with `1.3` is causing the problems since you've been looking at it?

**Comment by glaszig** [comment]
> what specifically with 1.3 is causing the problems

it was aruba 0.14 depending on cucumber >= 1.3.19.

```
>> bundle check --gemfile='/Users/martin/Code/capybara-screenshot/gemfiles/cucumber.1.3.0.gemfile' || bundle install --gemfile='/Users/martin/Code/capybara-screenshot/gemfiles/cucumber.1.3.0.gemfile'
Resolving dependencies...
Bundler can't satisfy your Gemfile's dependencies.
Install missing gems with `bundle install`.
Fetching gem metadata from http://rubygems.org/.............
Fetching version metadata from http://rubygems.org/..
Resolving dependencies...
Bundler could not find compatible versions for gem "cucumber":
  In snapshot (cucumber.1.3.0.gemfile.lock):
    cucumber (= 1.3.0)

  In Gemfile:
    cucumber (= 1.3.0)

    aruba (~> 0.14.0) was resolved to 0.14.0, which depends on
      cucumber (>= 1.3.19)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

my first impulse was locking aruba to a version compatible with cucumber 1.3.0.
but i just realize that simply upping the cucumber 1.3 appraisal to cucumber 1.3.19 and, voilá, tests are green. so, we can proof capybara-screenshot works with cucumber 1.3.19 and up.
if that's not ok for you, we'll have to downgrade aruba to < 0.8 and be back to square one -- incompatibility of aruba 0.14 test-code (`last_command_started`) with aruba 0.7.

**Comment by glaszig** [comment]
the root issue here is the `last_command_started` method introduced with newer aruba versions.
with cucumber 1.3 working as outlined above, we'll sill run into the same issue with cucumber 1.2.

either drop cucumber 1.2 or rewrite the tests to work with multiple aruba versions. yikes ;)

**Comment by glaszig** [comment]
i can't figure this one out. can you?

```
$ bundle exec appraisal spinach.0.7 rspec -b ./spec/cucumber/cucumber_spec.rb:71
>> BUNDLE_GEMFILE=/Users/martin/Code/capybara-screenshot/gemfiles/spinach.0.7.gemfile bundle exec rspec -b ./spec/cucumber/cucumber_spec.rb:71
Run options: include {:focus=>true, :locations=>{"./spec/cucumber/cucumber_spec.rb"=>[71]}}

Using Capybara::Screenshot with Cucumber
  pruning
    on failure it prunes previous screenshots when strategy is set (FAILED - 1)

Failures:

  1) Using Capybara::Screenshot with Cucumber pruning on failure it prunes previous screenshots when strategy is set
     Failure/Error: expect(last_command_started.output).to match(failure_message)

       expected "DEPRECATED: #save_and_open_page_path is deprecated, please use #save_path instead. \nNote: Behavior is slightly different with relative paths - see documentation\nno implicit conversion of Gherkin::TokenScanner into String (TypeError)\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:20:in `initialize'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:20:in `new'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:20:in `scan_setup'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:30:in `scan_str'\ngherkin.y:107:in `parse'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core/gherkin/parser.rb:30:in `document'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:27:in `block in parse'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:26:in `each'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:26:in `parse'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:18:in `compile'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-2.3.3/lib/cucumber/runtime.rb:68:in `run!'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-2.3.3/lib/cucumber/cli/main.rb:32:in `execute!'\n/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-2.3.3/bin/cucumber:8:in `<top (required)>'\n/Users/martin/.rbenv/versions/2.2.3/bin/cucumber:23:in `load'\n/Users/martin/.rbenv/versions/2.2.3/bin/cucumber:23:in `<main>'\n" to match "Unable to find link or button \"you'll never find me\""
       Diff:
       @@ -1,2 +1,19 @@
       -Unable to find link or button "you'll never find me"
       +DEPRECATED: #save_and_open_page_path is deprecated, please use #save_path instead.
       +Note: Behavior is slightly different with relative paths - see documentation
       +no implicit conversion of Gherkin::TokenScanner into String (TypeError)
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:20:in `initialize'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:20:in `new'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:20:in `scan_setup'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/gherkin-ruby-0.2.1/lib/gherkin/parser/lexer.rb:30:in `scan_str'
       +gherkin.y:107:in `parse'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core/gherkin/parser.rb:30:in `document'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:27:in `block in parse'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:26:in `each'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:26:in `parse'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-core-1.4.0/lib/cucumber/core.rb:18:in `compile'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-2.3.3/lib/cucumber/runtime.rb:68:in `run!'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-2.3.3/lib/cucumber/cli/main.rb:32:in `execute!'
       +/Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cucumber-2.3.3/bin/cucumber:8:in `<top (required)>'
       +/Users/martin/.rbenv/versions/2.2.3/bin/cucumber:23:in `load'
       +/Users/martin/.rbenv/versions/2.2.3/bin/cucumber:23:in `<main>'

     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-support-3.4.1/lib/rspec/support.rb:87:in `block in <module:Support>'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-support-3.4.1/lib/rspec/support.rb:96:in `call'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-support-3.4.1/lib/rspec/support.rb:96:in `notify_failure'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-expectations-3.4.0/lib/rspec/expectations/fail_with.rb:27:in `fail_with'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-expectations-3.4.0/lib/rspec/expectations/handler.rb:38:in `handle_failure'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-expectations-3.4.0/lib/rspec/expectations/handler.rb:50:in `block in handle_matcher'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-expectations-3.4.0/lib/rspec/expectations/handler.rb:27:in `with_matcher'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-expectations-3.4.0/lib/rspec/expectations/handler.rb:48:in `handle_matcher'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-expectations-3.4.0/lib/rspec/expectations/expectation_target.rb:54:in `to'
     # ./spec/cucumber/cucumber_spec.rb:11:in `run_failing_case'
     # ./spec/cucumber/cucumber_spec.rb:72:in `block (3 levels) in <top (required)>'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:236:in `instance_exec'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:236:in `block in run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:478:in `block in with_around_and_singleton_context_hooks'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:435:in `block in with_around_example_hooks'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/hooks.rb:478:in `block in run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/hooks.rb:618:in `block in run_around_example_hooks_for'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:320:in `call'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:320:in `call'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/aruba-0.14.1/lib/aruba/rspec.rb:23:in `block (2 levels) in <top (required)>'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:425:in `instance_exec'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:425:in `instance_exec'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/hooks.rb:389:in `execute_with'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/hooks.rb:620:in `block (2 levels) in run_around_example_hooks_for'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:320:in `call'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:320:in `call'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/hooks.rb:621:in `run_around_example_hooks_for'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/hooks.rb:478:in `run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:435:in `with_around_example_hooks'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:478:in `with_around_and_singleton_context_hooks'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example.rb:233:in `run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:581:in `block in run_examples'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:577:in `map'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:577:in `run_examples'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:543:in `run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:544:in `block in run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:544:in `map'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/example_group.rb:544:in `run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:119:in `block (3 levels) in run_specs'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:119:in `map'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:119:in `block (2 levels) in run_specs'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1680:in `with_suite_hooks'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:118:in `block in run_specs'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/reporter.rb:77:in `report'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:117:in `run_specs'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:93:in `run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:78:in `run'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:45:in `invoke'
     # /Users/martin/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/exe/rspec:4:in `<top (required)>'
     # /Users/martin/.rbenv/versions/2.2.3/bin/rspec:23:in `load'
     # /Users/martin/.rbenv/versions/2.2.3/bin/rspec:23:in `<main>'

Finished in 1.49 seconds (files took 0.75206 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/cucumber/cucumber_spec.rb:71 # Using Capybara::Screenshot with Cucumber pruning on failure it prunes previous screenshots when strategy is set
```

**Comment by glaszig** [comment]
spinach 0.8.2, specifically codegram/spinach@6776a40, fixes the above error. but don't ask me what actually makes the difference.

**Comment by mattheworiordan** [comment]
@glaszig sorry for the delay, I am really struggling to make time for `capybara-screenshot` lately 😦 

Given Spinach 0.8.3 came out Apr 2013, I would prefer that we simply stopped testing against this older version.  WDYT?

**Comment by glaszig** [comment]
let's drop spinach 0.7 and start support of the 0.8 branch from version 0.8.2.
i published what i've done so far in #175. let's continue the discussion there.
